test/support/src/vfs_helpers.cc

```diff
@@ -46,31 +46,23 @@
 #include "test/support/src/helpers.h"
 #include "test/support/src/vfs_helpers.h"
 
-namespace tiledb {
-namespace test {
+namespace tiledb::test {
 
 std::vector<std::unique_ptr<SupportedFs>> vfs_test_get_fs_vec() {
   std::vector<std::unique_ptr<SupportedFs>> fs_vec;
-
-  bool supports_s3 = false;
-  bool supports_hdfs = false;
-  bool supports_azure = false;
-  bool supports_gcs = false;
-  get_supported_fs(
-      &supports_s3, &supports_hdfs, &supports_azure, &supports_gcs);
-  if (supports_s3) {
+  if constexpr (tiledb::sm::filesystem::s3_enabled) {
```

Comment:
This has caused the cloud VFSes to be initialized even when I pass `--vfs native`, if TileDB is compiled with the cloud VFSes enabled.